src/checkers/inference/model/ConstraintManager.java

```diff
@@ -80,7 +81,10 @@ public SubtypeConstraint createSubtypeConstraint(Slot subtype, Slot supertype) {
 
             if (!constraintVerifier.isSubtype(subConstant, superConstant)) {
                 checker.report(Result.failure("subtype.constraint.unsatisfiable", subtype, supertype),
-                        visitorState.getPath().getLeaf());
+                        // visitorState.getPath().getLeaf() caused NullPointerException on some real project
+                        // and doesn't always return correct source tree. Using class tree at least shows which
+                        // Java source class there is such unsatisfiable error and also doesn't throw exception
+                        visitorState.getClassTree());
```

Comment:
How about making a small helper method that checks whether `visitorState.getPath()` is null and only if it is, uses the coarser-grain class tree?
Having the comment in the method would also make it obvious that the three other places also need to be changed once that issue is fixed.

Reply:
Done.